docs/architecture.md

```diff
@@ -71,9 +71,27 @@ Pay attention especially to the following sections:
 - **Architecture Invariant**: important note about crucial assumptions about the code/behavior
 - **API Boundary**: an exposed interface that requires careful design, e.g. because it has internal users to support
 
+### `ast`
+
+Types that serve as the Gleam AST, both in its *untyped* and *typed* form.
+Each tree describes a single Gleam module at its root.
+
+Untyped trees contain untyped expressions only.
+This form purely describes a sequence of characters.
```

Comment:
What does this mean?

Reply:
A bit clumsy attempt to say that this AST is free from semantic context, e.g. no additiona information about types, names, etc. If it's too unclear or redundant, we can remove this.